store/tikv/region_request.go

```diff
@@ -266,11 +267,26 @@ func (s *RegionRequestSender) SendReqCtx(
 	}
 
 	tryTimes := 0
+	mockRetrySendReqToRegionOnce := false
+	excludedAccessIdxes := make([]AccessIndex, 0)
 	for {
 		if (tryTimes > 0) && (tryTimes%1000 == 0) {
 			logutil.Logger(bo.GetCtx()).Warn("retry get ", zap.Uint64("region = ", regionID.GetID()), zap.Int("times = ", tryTimes))
 		}
 
+		// Stale Read request will retry the leader or next peer on error,
+		// so we will exclude the accessIdx of the requested store every time
+		// and claer it when all AccessIdxes are excluded to start a new round.
+		if rpcCtx != nil && req.GetStaleRead() {
+			opts = append(opts, WithoutAccessIdxes([]AccessIndex{rpcCtx.AccessIdx}))
+			if slice.AnyOf(excludedAccessIdxes, func(i int) bool {
+				return excludedAccessIdxes[i] == rpcCtx.AccessIdx
+			}) {
+				excludedAccessIdxes = make([]AccessIndex, 0)
+			} else {
+				excludedAccessIdxes = append(excludedAccessIdxes, rpcCtx.AccessIdx)
+			}
+		}
```

Comment:
I tend to let `onRegionError` returns the new `StoreSelectOption` in order to let stale read build a new `rpcContext` which we handle the logic in `onRegionError`

Reply:
Done. PTAL.